addons/mail/models/mail_message.py

```diff
@@ -1272,3 +1286,6 @@ def _invalidate_documents(self, model=None, res_id=None):
                     'message_needaction',
                     'message_needaction_counter',
                 ], ids=[res_id])
+
+    def _get_search_domain_share(self):
+        return ['&', ('is_internal', '=', False), ('subtype_id', '!=', False)]
```

Comment:
I haven't checked the call stack of this method in details, but it appears as if `subtype.internal != False` is missing from there.

Reply:
As I am back from copying internal from subtype to message, indeed.